### DISCUSSION TITLE
Calling Roles and Numbers from Collections

### DISCUSSION THREAD

**Discussion by dev-ref** [pocketbase/pocketbase#1681] [Q&A]
Hello, and apologies for what is probably an easy question in advance as I am a fairly new developer, but I need some advice on how to call on Roles and Numbers from a Collection.
For example in the Admin UI I have the 'users' collection. 
Within the collection I have the 'select' field labeled with the name 'role' and options 'A' and 'B'.
Second I have the 'number' field labeled with the name 'tokens'

I'm working in SvelteKit on functions that rely on which role the user has and a separate function that will update their tokens when called but I'm struggling to find the correct code to do so, specifically within {if} statements.
I've been able to set the API rules appropriately with things like `@request.auth.role = 'A' ` but as far as allowing access to certain links based on a user's role has proven difficult. If needed, I can supply broken code with what I am trying to accomplish if it would make more sense that way.

**Comment by ganigeorgiev** [comment]
I don't understand what do you mean by _"how to call on Roles and Numbers from a Collection"_.

Could you elaborate a little more on your data structure and what you are trying to do (code sample/pseudo-code would be also helpful)?

**Reply by dev-ref** [reply]
Some pseudo-code:

```
// Show link if user has been given the 'manager' role
{#if locals.pb.collection.('users').role = 'manager'}
	<a href="/manage">Management</a>
{/if}

// Display user's number of Tokens
{#each locals.pb.collection.('users').tokens as userTokens}
    <p>You have { userTokens } Tokens! </p>
{/each}
```

The fields in the Admin UI for each user includes a select field where they can be given a 'manager' role and a number field where the amount of 'tokens' they have will update accordingly. I'm not sure how to call those such as `{#each locals.pb.collection.('users').tokens as userTokens}` so that I can display that information through SvelteKit.

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand the example. If you want to iterate and access the record fields you'll have to first fetch them:

```svelte
<script>
    let users = [];

    loadUsers(); // or register it inside the onMount or loadData hook

    async function loadUsers() {
        // fetch all records at once
        users = await pb.collection('users').getFullList();
    }
</script>

{#each users as user (user.id)}
    {user.tokens}
{/each}
```

To access the currently authenticated user fields, you can call `pb.authStore.model.yourField`.

More information and examples how to use the SDK you can find in the [JS SDK README](https://github.com/pocketbase/js-sdk) and/or in the "Admin UI > {your collection} > API Preview".
